cypress/integration/console/shared/skip-payload-crypto/edit.spec.js

```diff
@@ -0,0 +1,250 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+describe('Skip payload crypto', () => {
+  const applicationId = 'spc-test-application'
+  const application = { ids: { application_id: applicationId } }
+  const userId = 'edit-app-spc-test-user'
+  const user = {
+    ids: { user_id: userId },
+    primary_email_address: 'edit-spc-test-user@example.com',
+    password: 'ABCDefg123!',
+    password_confirm: 'ABCDefg123!',
+  }
+
+  const endDeviceId = 'spc-test-end-device'
+  const endDevice = {
+    application_server_address: 'localhost',
+    ids: {
+      device_id: endDeviceId,
+      dev_eui: '0000000000000001',
+      join_eui: '0000000000000000',
+    },
+    name: 'End Device Test Name',
+    description: 'End Device Test Description',
+    join_server_address: 'localhost',
+    network_server_address: 'localhost',
+  }
+  const endDeviceFieldMask = {
+    paths: [
+      'join_server_address',
+      'network_server_address',
+      'application_server_address',
+      'ids.dev_eui',
+      'ids.join_eui',
+      'name',
+      'description',
+      'skip_payload_crypto_override',
+    ],
+  }
+  const endDeviceRequestBody = {
+    end_device: endDevice,
+    field_mask: endDeviceFieldMask,
+  }
+
+  before(() => {
+    cy.dropAndSeedDatabase()
+    cy.createUser(user)
+    cy.createApplication(application, userId)
+    cy.createEndDevice(applicationId, endDeviceRequestBody)
+  })
+
+  describe('Uplink', () => {
+    describe('application link skips payload crypto', () => {
+      beforeEach(() => {
+        cy.loginConsole({ user_id: userId, password: user.password })
+      })
+
+      it('disables messaging when not using a SPC overwrite', () => {
+        const adminApiKey = Cypress.config('adminApiKey')
+        const linkRequestBody = {
+          field_mask: { paths: ['skip_payload_crypto'] },
+          link: {
+            default_formatters: {},
+            skip_payload_crypto: true,
+          },
+        }
+        cy.request({
+          method: 'PUT',
+          url: `api/v3/as/applications/${applicationId}/link`,
+          body: linkRequestBody,
+          headers: {
+            Authorization: `Bearer ${adminApiKey}`,
+          },
+        })
+
+        const response = {
+          skip_payload_crypto_override: null,
+          session: {},
+        }
+
+        cy.intercept(
+          'GET',
+          `as/applications/${applicationId}/devices/${endDeviceId}?field_mask=version_ids,formatters,skip_payload_crypto_override,session,pending_session`,
+          response,
+        )
```

Comment:
Only use intercepts if absolutely necessary. In the end-to-end tests we try to reproduce usage as closely as possible, this includes relying on actual API responses wherever possible. 
In this case, you could either actually update the end device using a `PUT` request, or initially create the end device with the desired configuration directly.